### PR TITLE
Upgrade random access storage

### PR DESCRIPTION
**Pull request by khernyo**
## Checklist

- [x] tests pass

## Semver Changes
Minor

### CODE REVIEW

**Comment by khernyo** [comment]
This is an incomplete PR, just to show how it would look. A new version is required from all the `random-access-*` crates to make it work. A couple of small changes are also needed in `random-access-memory` and `random-access-disk`, but nothing important.

Is this what you had in mind?

To be honest, I don't think this is the right move. I think I understand why you would want to abstract over the error type, but I would prefer if `random-access-storage` used the failure crate. The rust community seems to be moving toward using the failure crate for error handling and also integrating parts of it into std. It provides backtrace and a chain of causes, it can be used with `no_std` and everybody is (or will be) familiar with it. Also, `random-access-storage` could be used without the `<Error = Error>` constraint which pretty much forbids any other error types if used with hypercore.

Anyway, if this is how you want to go forward then I'm happy to open the necessary PRs in the `random-access-*` projects to make this PR work.

File: Cargo.toml [hunk 1]
```diff
@@ -19,9 +19,9 @@ memory-pager = "0.4.5"
 merkle-tree-stream = "0.5.0"
 pretty-hash = "0.2.0"
 rand = "0.4.2"
-random-access-disk = "0.4.0"
-random-access-memory = "0.3.0"
-random-access-storage = "0.4.0"
+random-access-disk = { path = "../random-access-disk" }
+random-access-memory = { path = "../random-access-memory" }
+random-access-storage = { path = "../random-access-storage" }
```
Comment (yoshuawuyts) [inline Cargo.toml]: We can't land this, as it'll prevent us from cutting releases

**Comment by yoshuawuyts** [comment]
> (...) but I would prefer if random-access-storage used the failure crate (...)

We used to do this, but this prevented us from defining our own error types upstream. E.g. for [Error + ErrorKind pattern](https://boats.gitlab.io/failure/error-errorkind.html) to work, the `RandomAccessStorage` trait needs to be generic over the error.

But these changes do look good; I like it!

**Comment by khernyo** [comment]
Did you mean to push the "Update .github" commit onto this branch or was it by accident? You reverted some of my changes, but only some of them, so it doesn't make sense to me.

The ISSUE_TEMPLATE changes can stay, I don't mind (though I don't think those belong here).

**Comment by khernyo** [comment]
Could you please release a new random-access-memory and random-access-disk version at the current master? This PR should work with those (after updating the versions in Cargo.toml).

Also, I noticed that you created similar "Update .github" commits in all "random-access-*" repos, and all of them seem to be pushed to old, already merged branches.

**Comment by yoshuawuyts** [comment]
> Also, I noticed that you created similar "Update .github" commits in all "random-access-*" repos, and all of them seem to be pushed to old, already merged branches.

Yeah oops, I ran a batch script but forgot to checkout master first haha. Oops oops.

> Did you mean to push the "Update .github" commit onto this  branch or was it by accident? You reverted some of my changes, but only  some of them, so it doesn't make sense to me.

I did not!! Sorry about that!

**Comment by yoshuawuyts** [comment]
Published new versions of `random-access-disk` and `random-access-memory`!

**Comment by khernyo** [comment]
Thanks for the releases and no worries about the runaway script :)

It's almost ready to go now, I just need to sign-off on the commit (which I will do later today).

**Comment by yoshuawuyts** [comment]
Okay lol bors isn't going to work now. Merging manually!

**Pull request merged**